app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -2,32 +2,93 @@ package com.andrew.tasky.agenda.data.agenda
 
 import android.util.Log
 import com.andrew.tasky.agenda.data.database.AgendaDatabase
-import com.andrew.tasky.agenda.data.reminder.ReminderApi
 import com.andrew.tasky.agenda.data.util.*
 import com.andrew.tasky.agenda.domain.AgendaRepository
 import com.andrew.tasky.agenda.domain.ReminderRepository
 import com.andrew.tasky.agenda.domain.models.AgendaItem
 import com.andrew.tasky.auth.data.AuthResult
 import com.andrew.tasky.auth.util.getAuthResult
 import java.time.LocalDate
+import java.time.LocalDateTime
 import java.time.LocalTime
 import java.util.*
+import kotlinx.coroutines.*
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.map
 
 class AgendaRepositoryImpl(
     private val agendaApi: AgendaApi,
     private val reminderRepository: ReminderRepository,
     private val db: AgendaDatabase,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
 ) : AgendaRepository {
 
-    override fun getAgendaItems(): Flow<List<AgendaItem>> {
-        val localReminders = db.getReminderDao().getRemindersAsFlow().map {
+    override fun getAgendaItems(localDate: LocalDate): Flow<List<AgendaItem>> {
+
+        val calendar = Calendar.getInstance()
+        calendar.set(
+            localDate.year,
+            localDate.monthValue - 1,
+            localDate.dayOfMonth
+        )
+        calendar.set(Calendar.HOUR_OF_DAY, 0)
+        calendar.set(Calendar.MINUTE, 0)
+        calendar.set(Calendar.SECOND, 0)
+        calendar.set(Calendar.MILLISECOND, 0)
+        val startEpochMilli = calendar.timeInMillis
+
+        calendar.add(Calendar.DATE, 1)
+        val endEpochMilli = calendar.timeInMillis
```

Comment:
You don't really need `Calendar` for anything, it's just a lot of boilerplate code here. You can do this all with `LocalDateTime`/ `ZonedDateTime` more convenient and with less code